pkg/statedb/table.go

```diff
@@ -194,13 +194,41 @@ func (t *genTable[Obj]) DeleteAll(txn WriteTxn) error {
 	return nil
 }
 
-func (t *genTable[Obj]) DeleteTracker(txn WriteTxn, trackerName string) (*DeleteTracker[Obj], error) {
+func (t *genTable[Obj]) DeleteTracker(txn ReadTxn, trackerName string) (*DeleteTracker[Obj], error) {
+	// To allow DeleteTracker() to be used by readers, we need to deal with these cases:
+	//
+	// 1. txn is just a ReadTxn, no tables locked => new WriteTxn
+	// 2. txn is a WriteTxn, but it's not locking the target table => new WriteTxn
+	// 3. txn is a WriteTxn and it's locking the target table => reuse txn
+	//
+	// There's one unfortunate case still, where the given 'txn' is a ReadTxn, but the caller
+	// has created a WriteTxn that is still open and targets this table. This will lead to a
+	// deadlock and there isn't much we can do about that. DeleteTracker() in Table[Obj] documents
+	// this caviat.
```

Comment:
nit, spelling: caveat (also in commit msg)

Reply:
This is gone now.